adapter/src/test/kotlin/de/fraunhofer/iem/spha/adapter/kpis/cve/CveAdapterTest.kt

```diff
@@ -9,77 +9,58 @@
+        assertEquals(50, kpi2?.score)
+
+        val kpi3 = transformVulnerabilityToKpi(score = 10.0, typeId = KpiType.NUMBER_OF_COMMITS)
+        assertEquals(0, kpi3?.score)
+
+        // Test with originId
+        val originId = "test-origin-id"
+        val kpi4 =
+            transformVulnerabilityToKpi(
+                score = 7.5,
+                typeId = KpiType.NUMBER_OF_COMMITS,
+                originId = originId,
+            )
+        assertEquals(25, kpi4?.score)
+        assertEquals(originId, kpi4?.originId)
+    }
+
     @Test
     fun basicVulnerabilityToKpiTransformation() {
         // valid input
-        val validKpi =
-            CveAdapter.transformCodeVulnerabilityToKpi(
-                listOf(
-                    VulnerabilityDto(
-                        cveIdentifier = "not blank",
-                        packageName = "not blank",
-                        severity = 0.1,
-                        version = "0.0.1",
-                    )
-                )
-            )
-        when (val kpi = validKpi.first()) {
-            is AdapterResult.Success<*> -> {
-                assert(kpi.rawValueKpi.score in (0..100))
-            }
+        val validKpi = transformVulnerabilityToKpi(score = 7.5, typeId = KpiType.NUMBER_OF_COMMITS)
 
-            // no type parameters needed here
-            else -> fail()
+        if (validKpi == null || validKpi.score != 25) {
+            fail()
         }
 
         // invalid input
         val invalidKpis =
-            CveAdapter.transformCodeVulnerabilityToKpi(
-                listOf(
-                    VulnerabilityDto(
-                        cveIdentifier = "not blank",
-                        packageName = "",
-                        severity = 0.1,
-                        version = "0.0.1",
-                    ),
-                    VulnerabilityDto(
-                        cveIdentifier = "",
-                        packageName = "not blank",
-                        severity = 0.1,
-                        version = "0.0.1",
-                    ),
-                    VulnerabilityDto(
-                        cveIdentifier = "not blank",
-                        packageName = "not blank",
-                        severity = -0.1,
-                        version = "0.0.1",
-                    ),
-                    VulnerabilityDto(
-                        cveIdentifier = "not blank",
-                        packageName = "not blank",
-                        severity = 10.1,
-                        version = "0.0.1",
-                    ),
-                )
+            listOf(
+                transformVulnerabilityToKpi(score = -7.5, typeId = KpiType.NUMBER_OF_COMMITS),
+                transformVulnerabilityToKpi(score = 11.5, typeId = KpiType.NUMBER_OF_COMMITS),
+                transformVulnerabilityToKpi(score = 10.1, typeId = KpiType.NUMBER_OF_COMMITS),
```

Comment:
Minor, from testing perspective you tested for a value above but not right below, just suggesting it for next time.
```suggestion
                transformVulnerabilityToKpi(score = -0.1, typeId = KpiType.NUMBER_OF_COMMITS),
                transformVulnerabilityToKpi(score = 10.1, typeId = KpiType.NUMBER_OF_COMMITS),
```